specs/marketplace.md

```diff
@@ -116,9 +89,22 @@ A client has two primary responsibilities:
 
 When a user prompts the client node to create a storage request, the client node SHOULD receive the input parameters for the storage request from the user.
 
-To create a request to persist a dataset on the Codex network, client nodes MUST split the dataset into data chunks, $(c_1, c_2, c_3, \ldots, c_{n})$. Using the erasure coding method and the provided input parameters, the data chunks are encoded and distributed over a number of slots. The applied erasure coding method MUST use the [Reed-Solomon algorithm](https://hackmd.io/FB58eZQoTNm-dnhu0Y1XnA). The final slot roots and other metadata MUST be placed into a `Manifest` (TODO: Manifest RFC). The CID for the `Manifest` MUST then be used as the `cid` for the stored dataset.
+To create a request to persist a dataset on the Codex network, client nodes MUST split the dataset into data chunks, $(c_1, c_2, c_3, \ldots, c_{n})$.
+The data chunks are then encoded with the erasure coding method and the provided input parameters.
+
+The applied erasure coding method, used on data being persisted all SPs,
+SHOULD be the [Reed-Solomon algorithm](https://hackmd.io/FB58eZQoTNm-dnhu0Y1XnA).
+If the client node uses a different algorithm, once storage request is made,
+SP will not be able to submit valid proofs.
+
```

Comment:
Bring this to discussion once more because I am still not sure it should be MUST here. Does the smart contract reject a request when data is not erasure encoded, or do SP check when data is downloaded? I believe the `MUST` is required when the validation occurs, which is mentioned later in the spec.

Reply:
Ok, will keep at `MUST`.  I also changed the last statement to include " When dataset is encoded with different algorithm, the dataset will be inaccessible."

Reply:
Sounds good - thanks